python/tk_maya/menu_generation.py

```diff
@@ -197,19 +200,56 @@ def _add_app_menu(self, commands_by_app):
                 cmd_obj = commands_by_app[app_name][0]
                 if not cmd_obj.favourite:
                     # skip favourites since they are already on the menu
-                    cmd_obj.add_command_to_menu(self._menu_handle)
+                    cmd_obj.add_command_to_menu(self._menu_path)
+
+
+class Callback(object):
```

Comment:
Broken the callback logic out of the AppCommand class so that this can be used directly by the jump to filesystem and 
Shotgun commands in place of the pymel Callback class.

AppCommand now derives from Callback.

Reply:
Should this `Callback` class be refactored into it's own file? It looks pretty much the same to me.

Reply:
They are the same, the trouble is, that one is defined in the plugin startup code (`plugins/basic/python/tk_maya_basic`), and the other in the menu generation code (`python/tk_maya/menu_generation.py`).
There isn't a precedent so far for importing files from one location to the other. If we were to refactor it, it would make sense to store it in the `python/tk-maya` folder and then import it from there in the plugin launch code as well as the menu_generation.
I could refactor it, and then I guess add the python/tk-maya path to the sys path so I can import it, but I'm not sure if that is really a good approach.